src/dbnode/storage/database.go

```diff
@@ -317,14 +310,40 @@ func (d *db) UpdateOwnedNamespaces(newNamespaces namespace.Map) error {
 			"restart the process if you want changes to take effect")
 	}
 
-	// enqueue bootstraps if new namespaces
 	if len(adds) > 0 {
-		d.queueBootstrapWithLock()
```

Comment:
Isn't there potentially the same problem in another use case of `queueBootstrapWithLock` (when new shards are being transferred to the node)?

Reply:
I reduced the scope of db locking in `assignShardSet()` as well. Tested node bootstraps and they didn't have any issues.

Reply:
Updated logics in `assignSharSet()`.